src/hooks/use-content-folders.ts

```diff
@@ -0,0 +1,60 @@
+import * as Sentry from '@sentry/electron/renderer';
+import { useState, useEffect } from 'react';
+import { getIpcApi } from 'src/lib/get-ipc-api';
+
+export type WpContentFolder = 'plugins' | 'themes';
+export type WpContentFolderEntry = { name: string; type: 'file' | 'folder' };
+export type WpContentFolderResult = {
+	items: WpContentFolderEntry[];
+	isLoading: boolean;
+	error: Error | null;
+};
+
+export function useContentFolders(
+	siteId: string,
+	types: WpContentFolder[]
```

Comment:
I'm not sure if we need to pass types. I like the flexibility of the hook, but I think it's not necessary, and useContentFolders can always return the plugins and themes children. Let's reconsider if we need pas the types parameter or not on STU-580.